### PR TITLE
Refactor some rotation matrix construction code

### PR DESCRIPTION
**Pull request by markdewing**
Separate some of the rotation matrix code into pieces that depend on the object, and some that are independent of the class and are easier to test in isolation.

- Create a type alias (RotationIndices) for the vector of pairs that stores the indices for active rotation parameters.
- Extract code to createRotationIndices and constructAntiSymmetricMatrix and make them static functions.
- Add a few tests for these new functions

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by markdewing** [comment]
Test this please

File: src/QMCWaveFunctions/RotatedSPOs.cpp [hunk 1]
```diff
@@ -37,6 +37,34 @@ void RotatedSPOs::setRotationParameters(const std::vector<RealType>& param_list)
   params_supplied = true;
 }
 
+// Construct a list of the matrix indices for non-zero rotation parameters
+//  (The structure for a sparse representation of the matrix)
+// Only core->active rotations are created.
+void RotatedSPOs::createRotationIndices(int nel, int nmo, RotationIndices& rot_indices)
```
Comment (ye-luo) [inline src/QMCWaveFunctions/RotatedSPOs.cpp:43]: It think the comments are better with the function declaration. If they are better for the function definition, comment lines can be better with the actual code.

File: src/QMCWaveFunctions/RotatedSPOs.cpp [hunk 2]
```diff
@@ -37,6 +37,34 @@ void RotatedSPOs::setRotationParameters(const std::vector<RealType>& param_list)
...
+{
+  for (int i = 0; i < nel; i++)
+    for (int j = nel; j < nmo; j++)
+      rot_indices.push_back(std::pair<int, int>(i, j));
```
Comment (ye-luo) [inline src/QMCWaveFunctions/RotatedSPOs.cpp:47]: More compact with `rot_indices.emplace_back(i,j)`

File: src/QMCWaveFunctions/RotatedSPOs.cpp (hunk 1 above)
Comment (prckent) [reply src/QMCWaveFunctions/RotatedSPOs.cpp:40]: Yes, this is our standard practice.

**Review by prckent — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**